packages/amplify-graphql-api-construct/src/internal/construct-exports.ts

```diff
@@ -56,9 +57,14 @@ export const getGeneratedResources = (scope: Construct): AmplifyGraphqlApiResour
       return;
     }
 
-    // Retrieve reference name for indexed resources, and bail if none is found.
+    // Retrieve reference name for indexed resources, store resources in fallback if name not found
     const resourceName = getResourceName(currentScope);
-    if (!resourceName) return;
+    if (!resourceName) {
+      if (currentScope instanceof CfnResource) {
+        additionalCfnResourcesWithoutResourceName.push(currentScope);
+      }
+      return;
+    }
```

Comment:
Is this just added for testing? I'd prefer we avoid doing this unless we really need it for some reason. If a customer wants to walk the full construct tree they can always do something similar to what we have in this utility.